src/Container/AgFeatureGrid/AgFeatureGrid.jsx

```diff
@@ -0,0 +1,1021 @@
+      this.initVectorLayer(nextProps.map);
+      this.initMapEventHandlers(nextProps.map);
+    }
+
+    if (!(isEqual(nextProps.features, features))) {
+      if (this._source) {
+        this._source.clear();
+        this._source.addFeatures(nextProps.features);
+      }
+
+      if (nextProps.zoomToExtent) {
+        this.zoomToFeatures(nextProps.features);
+      }
+    }
+
+    if (!(isEqual(nextProps.selectable, selectable))) {
+      if (nextProps.selectable && map) {
+        map.on('singleclick', this.onMapSingleClick);
+      } else {
+        if (map) {
+          map.un('singleclick', this.onMapSingleClick);
+        }
+      }
+    }
+  }
+
+  /**
+   * Called on lifecycle phase componentWillUnmount.
+   */
+  componentWillUnmount() {
+    this.deinitVectorLayer();
+    this.deinitMapEventHandlers();
+  }
+
+  /**
+   * Initialized the vector layer that will be used to draw the input features
+   * on and adds it to the map (if any).
+   *
+   * @param {ol.Map} map The map to add the layer to.
+   */
+  initVectorLayer = map => {
+    const {
+      features,
+      featureStyle,
+      layerName
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    if (MapUtil.getLayerByName(map, layerName)) {
+      return;
+    }
+
+    const source = new OlSourceVector({
+      features: features
+    });
+
+    const layer = new OlLayerVector({
+      name: layerName,
+      source: source,
+      style: featureStyle
+    });
+
+    map.addLayer(layer);
+
+    this._source = source;
+    this._layer = layer;
+  }
+
+  /**
+   * Adds map event callbacks to highlight and select features in the map (if
+   * given) on pointermove and singleclick. Hovered and selected features will
+   * be highlighted and selected in the grid as well.
+   *
+   * @param {ol.Map} map The map to register the handlers to.
+   */
+  initMapEventHandlers = map => {
+    const {
+      selectable
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    map.on('pointermove', this.onMapPointerMove);
+
+    if (selectable) {
+      map.on('singleclick', this.onMapSingleClick);
+    }
+  }
+
+  /**
+   * Highlights the feature beneath the cursor on the map and in the grid.
+   *
+   * @param {ol.MapBrowserEvent} olEvt The ol event.
+   */
+  onMapPointerMove = olEvt => {
+    const {
+      map,
+      features,
+      highlightStyle,
+      selectStyle
+    } = this.props;
+
+    const {
+      grid
+    } = this.state;
+
+    const selectedRowKeys = this.getSelectedRowKeys();
+
+    const selectedFeatures = map.getFeaturesAtPixel(olEvt.pixel, {
+      layerFilter: layerCand => layerCand === this._layer
+    }) || [];
+
+    const rowRenderer = grid.api.rowRenderer;
+
+    features.forEach(feature => {
+      const key = this.props.keyFunction(feature);
+
+      let rc;
+      rowRenderer.forEachRowComp((idx, rowComp) => {
+        if (rowComp.getRowNode().data.key === feature.ol_uid) {
```

Comment:
This will only work for the default `keyFunction` doesn't it? I think it should be `=== key`

Reply:
Nice catch! Fixed in 85cacd7 .